service/db_access.py

```diff
@@ -1,7 +1,36 @@
 from sqlalchemy import false                                 # type: ignore
 from sqlalchemy.orm.strategy_options import load_only, Load  # type: ignore
+from service import db
+from service.models import TitleRegisterData, UprnMapping, UserSearchAndResults
 
-from service.models import TitleRegisterData, UprnMapping
+
+def save_user_search_details(params):
+    """
+    Save user's search request details, for audit purposes.
+    """
+
+    user_search_request = UserSearchAndResults\
+                            (
+                            viewed_time=params['MC_timestamp'],
```

Comment:
need to create a new 'insert time' or similar name to part of the compound key
need to change the 'viewed_time' row to be nullable as it may never be completed